app/carbonapi/api.go

```diff
@@ -245,42 +247,42 @@ func registerPromehteusMetrics(logger *zap.Logger) {
 	}()
 }
 
-func loadBlockRuleHeaderConfig(ticker *time.Ticker, logger *zap.Logger) {
+func loadBlockRuleHeaderConfig(ticker *time.Ticker, logger *zap.Logger, app *App) {
 	var ruleConfig RuleConfig
 	for range ticker.C {
-		fileData, err := loadBlockRuleConfig()
+		fileData, err := loadBlockRuleConfig(app)
 
 		if err == nil {
 			err = yaml.Unmarshal(fileData, &ruleConfig)
 			if err != nil {
 				logger.Error("couldn't unmarshal block rule file data")
 			} else {
-				config.blockHeaderRules = ruleConfig
+				app.blockHeaderRules = ruleConfig
```

Comment:
I think if a func is mutating the `app` struct, it should probably be a method hanging off that struct, rather than a function which takes the struct as an argument.

Also, I'm not sure about the context here, but now that our HTTP handlers are calling methods on the struct, we need to be really paranoid about any code which mutates fields on the app struct, since they can represent a data race.